integration/spark/spark3/src/main/java/io/openlineage/spark3/agent/lifecycle/plan/catalog/DatabricksUnityV2Handler.java

```diff
@@ -0,0 +1,113 @@
+/*
+/* Copyright 2018-2022 contributors to the OpenLineage project
+/* SPDX-License-Identifier: Apache-2.0
+*/
+
+package io.openlineage.spark3.agent.lifecycle.plan.catalog;
+
+import io.openlineage.client.OpenLineage;
+import io.openlineage.spark.agent.util.DatasetIdentifier;
+import io.openlineage.spark.agent.util.PathUtils;
+import io.openlineage.spark.api.OpenLineageContext;
+import java.lang.reflect.InvocationTargetException;
+import java.util.Arrays;
+import java.util.Map;
+import java.util.Optional;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.reflect.MethodUtils;
+import org.apache.hadoop.fs.Path;
+import org.apache.spark.sql.SparkSession;
+import org.apache.spark.sql.catalyst.TableIdentifier;
+import org.apache.spark.sql.connector.catalog.Identifier;
+import org.apache.spark.sql.connector.catalog.TableCatalog;
+import scala.Option;
+
+/**
+ * The DatabricksUnityV2Handler is intended to support Databricks' custom Unity Catalog which has
+ * the class name of com.databricks.sql.managedcatalog.UnityCatalogV2Proxy rather than the open
+ * source class name of org.apache.spark.sql.delta.catalog.DeltaCatalog. It is used in the same way
+ * as the {@link DeltaHandler}.
+ */
+@Slf4j
+public class DatabricksUnityV2Handler implements CatalogHandler {
+
+  private final OpenLineageContext context;
+
+  public DatabricksUnityV2Handler(OpenLineageContext context) {
+    this.context = context;
+  }
+
+  @Override
+  public boolean hasClasses() {
+    try {
+      DeltaHandler.class
+          .getClassLoader()
+          .loadClass("com.databricks.sql.managedcatalog.UnityCatalogV2Proxy");
+      return true;
+    } catch (Exception e) {
+      // swallow- we don't care
+    }
+    return false;
+  }
+
+  @Override
+  public boolean isClass(TableCatalog tableCatalog) {
+    return "com.databricks.sql.managedcatalog.UnityCatalogV2Proxy"
+        .equals(tableCatalog.getClass().getCanonicalName());
+  }
+
+  @Override
+  public DatasetIdentifier getDatasetIdentifier(
```

Comment:
If it is a copy/paste from `DatabricksDeltaHandler` (as described in PR), please consider creating an `AbstractDatabricksHandler` class with `isClass` and `getDatasetIdentifier` methods implemented and let classes `DatabricksUnityV2Handler` and `DatabricsDeltaHandler` extend this abstract class.

Reply:
@pawel-big-lebowski can do.  My only concern is that we have no idea how Unity is going to evolve.  Is it still worth handling if Unity suddenly grows to be a completely different beast than the way Databricks Delta catalog was being handled?

Reply:
We make use of non-public API. There is no guarantee on how do they evolve and there is little we can do about that. For the future versions we can create new Handlers within `spark32`, `spark33`.... projects.

Reply:
Got it!  Thank you so much for the guidance, Pawel!  I really appreciate the perspective and assurance that we can handle this as they grow.

Reply:
@pawel-big-lebowski  per your suggestion, I've refactored this to use an `AbstractDatabricksHandler` instead and noted their relationships.